orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -93,8 +93,8 @@ class TaskController {
 
   @RequestMapping(value = "/tasks/{id}/cancel", method = RequestMethod.PUT)
   @ResponseStatus(HttpStatus.ACCEPTED)
-  void cancelTask(@PathVariable String id) {
-    executionRepository.cancel(id)
+  void cancelTask(@PathVariable String id, @RequestParam(value = "user", required = false) String user) {
```

Comment:
We don't/shouldn't be passing 'user' in as a request parameter.

I reckon you could do `AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")`.

That's what we do in StageBuilder to track who restarted a pipeline (also do it from TaskController for tracking lastModifiedBy).